runtime/src/main/java/org/corfudb/runtime/view/replication/QuorumReplicationProtocol.java

```diff
@@ -292,9 +291,7 @@ private ReadResponse getAdoptedValueWithHighestRankIfPresent(
         public int compare(ReadResponse o1, ReadResponse o2) {
             LogData ld1 = o1.getAddresses().get(logPosition);
             LogData ld2 = o2.getAddresses().get(logPosition);
-            if(ld1.isTrimmed() || ld2.isTrimmed()) {
-                throw new TrimmedException();
-            }
+
```

Comment:
I'm not very sure of the logic here but based on what we removed, shouldn't we check ifs either log data has been compacted? or will the rank information still be available even if it was compacted?

Reply:
I also noticed this problem and addressed in the most recent PR.